test/graaflib/algorithm/shortest_path_test.cpp

```diff
@@ -410,4 +411,86 @@ TYPED_TEST(DijkstraShortestPathTest, DijkstraMoreComplexShortestPathTree) {
   ASSERT_EQ(path_map, expected_path_map);
 }
 
+template <typename T>
+struct DijkstraShortestPathSignedTypesTest : public testing::Test {
+  using graph_t = typename T::first_type;
+  using edge_t = typename T::second_type;
+};
+
+using weighted_graph_signed_types = testing::Types<
```

Comment:
I don't see a much better solution than this 👍🏻 

In a follow-up I plan to rethink our testing code and see if I can extract some common patterns. For instance I think it would be sufficient to have 2 or 3 such test fixture definitions which we can reuse through the entire testing code. 

Furthermore it might be nice to have some common pre-defined test structures we ca re-use for testing the various algorithms.

If you have any more ideas/pain points I would be happy to hear them!